app/src/test/java/org/simple/clinic/sync/indicator/SyncIndicatorViewControllerTest.kt

```diff
@@ -124,26 +125,34 @@ class SyncIndicatorViewControllerTest {
   }
 
   @Test
-  @Parameters(method = "params for testing sync errors")
-  fun `when sync indicator is clicked and sync starts, appropriate failure dialog should show if any sync error is thrown`(error: ResolvedError) {
-    whenever(dataSync.streamSyncErrors()).thenReturn(Observable.just(error))
+  fun `appropriate failure dialog should be shown for all sync errors except Unauthorized`() {
```

Comment:
Can we split this into 4 different tests?

Reply:
Sorry, this change can be reverted. Because this is an ideal case for a parameterized test.

Reply:
Works!

Reply:
Done.